emr-dynamodb-hadoop/src/main/java/org/apache/hadoop/dynamodb/read/AbstractDynamoDBInputFormat.java

```diff
@@ -65,13 +70,26 @@ public InputSplit[] getSplits(JobConf conf, int desiredSplits) throws IOExceptio
     }
 
     long tableSizeBytes = conf.getLong(DynamoDBConstants.TABLE_SIZE_BYTES, 1);
-    int numSegments = getNumSegments(configuredReadThroughput, (int)
+    int numTotalSegments = getNumSegments(configuredReadThroughput, (int)
         maxWriteThroughputAllocated, tableSizeBytes, conf);
-    int numMappers = getNumMappers(numSegments, configuredReadThroughput, conf);
 
-    log.info("Using " + numSegments + " segments across " + numMappers + " mappers");
+    Set<Integer> excludedSegments =
+        Arrays.stream(conf.getInts(DynamoDBConstants.EXCLUDED_SCAN_SEGMENTS))
+            .boxed()
+            .collect(Collectors.toSet());
 
-    return getSplitGenerator().generateSplits(numMappers, numSegments, conf);
+    List<Integer> segments =
+        IntStream.range(0, numTotalSegments)
+            .boxed()
+            .filter(i -> !excludedSegments.contains(i))
```

Comment:
will this work even if there are overlaps or subsets?

in CQL you have token ranges, which are sized by split count, but algorithm counts with that


are we sure here that if e.g. input params (such as number of mappers or read throughput) change that this algorithm will still work?

or we mandate same input params?